core/test/regression/site/site_spec.js

```diff
@@ -2625,10 +2625,28 @@ describe('Integration - Web - Site', function () {
                 });
             });
 
-            describe('collection with data key', function () {
+            describe('collection/routes with data key', function () {
                 before(function () {
                     sinon.stub(settingsService, 'get').returns({
-                        routes: {},
+                        routes: {
+                            '/my-page/': {
+                                data: {
+                                    query: {
+                                        page: {
+                                            controller: 'pagesPublic',
+                                            resource: 'pages',
+                                            type: 'read',
+                                            options: {
+                                                slug: 'static-page-test'
+                                            }
+                                        }
+                                    },
+                                    router: {
+                                        pages: [{redirect: true, slug: 'static-page-test'}]
+                                    }
+                                }
+                            }
```

Comment:
The use case was not covered in dynamic routing regression test. Now it's there.
It errors on master

Reply:
🤓

Reply:
Hm.. Tried it on master, didn't error :thinking:

Reply:
It hangs!

Reply:
You were right :+1: my :b: